pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -811,6 +812,11 @@ public void connectionOpened(final ClientCnx cnx) {
         if (si != null && (SchemaType.BYTES == si.getType() || SchemaType.NONE == si.getType())) {
             // don't set schema for Schema.BYTES
             si = null;
+        } else {
+            if (schema instanceof AutoConsumeSchema
+                    && Commands.peerSupportsCarryAutoConsumeSchemaToBroker(cnx.getRemoteEndpointProtocolVersion())) {
```

Comment:
if the autoConsumeSchema has the `SchemaVersion.Latest` schema, maybe we can't set the `si = ((AutoConsumeSchema) schema).getAutoConsumeSchemaInfo()`,

Reply:
Yes, yes, but you can look at it.
`SchemaInfo si = schema.getSchemaInfo();`
And
https://github.com/apache/pulsar/blob/06e4db5c821b2ba9241040f5db52f882e83e8cd8/pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AutoConsumeSchema.java#L140-L146
So `getAutoConsumeSchemaInfo` is executed when si is null.